### PR TITLE
LPS-205412 Calculate pageContents only when needed

### PR DESCRIPTION
**Pull request by sandrodw3**

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Review by lfbesada — APPROVED** [review]
@sandrodw3 Thank you very much for the PR, it works great, and from the backend point of view LGTM :star_struck: 
I'm going to launch the relevant tests because I suspect that some integration tests will have to be adapted.
I will be waiting to adapt those failed.

**Comment by lfbesada** [comment]
ci:test:relevant

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/hooks/app_hooks/usePortletConfigurationListener.js [hunk 1]
```diff
@@ -24,15 +23,11 @@ export default function usePortletConfigurationListener() {
...
-						updatePageContents({
-							pageContents,
-						})
-					);
-				});
+				dispatch(
+					updatePageContents({
+						pageContents: null,
+					})
+				);
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/hooks/app_hooks/usePortletConfigurationListener.js:30]: Maybe we could split this action into "discardPageContents" and "setPageContents", something more explicit than setting them to null. What do you think about that? Maybe too over-engineered?

File: modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/utils/usePageContents.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {useEffect, useState} from 'react';
+
+import updatePageContents from '../actions/updatePageContents';
+import {useDispatch, useSelector} from '../contexts/StoreContext';
+import {selectPageContents} from '../selectors/selectPageContents';
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-content-page-editor-web/src/main/resources/META-INF/resources/page_editor/app/utils/usePageContents.js:10]: We can delete the selectPageContents selector (or inline it here), it shouldn't be used anywhere else.

**Review by p2kmgcl — APPROVED** [review]
Awesome :smile: 

Did you have the opportunity to test if there is some performance improvement?

**Comment by lfbesada** [comment]
ci:report:2488741412

**Comment by sandrodw3** [comment]
ci:test:sf

**Comment by sandrodw3** [comment]
ci:test:relevant

**Comment by sandrodw3** [comment]
@p2kmgcl Applied the changes we've been discussing 😄 adding Frontend label again just in case you want to take another look. Also added @lfbesada's fix for integrations tests. We probably still have to do something with poshi 😭

**Comment by p2kmgcl** [comment]
@sandrodw3 awesome :smile:

**Comment by p2kmgcl** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:test:relevant

**Comment by p2kmgcl** [comment]
ci:forward

**Comment by sandrodw3** [comment]
ci:report:2530159478

**Comment by ealonso** [comment]
ci:test:relevant

**Comment by sandrodw3** [comment]
ci:report:2530465232

**Comment by sandrodw3** [comment]
ci:test:relevant

**Comment by sandrodw3** [comment]
ci:stop

**Comment by sandrodw3** [comment]
ci:test:relevant

**Comment by sandrodw3** [comment]
ci:report:2530632127

**Comment by sandrodw3** [comment]
ci:forward

**Comment by sandrodw3** [comment]
ci:report:2530812634

**Comment by sandrodw3** [comment]
ci:forward

**Comment by sandrodw3** [comment]
ci:forward:force

**Pull request closed without merging**